js/app.js

```diff
@@ -84,18 +85,42 @@ const App = {
 			}
 		}
 	},
+	async login() {
+		// Login removes the document fragment, so keep it in storage
+		// See https://www.rfc-editor.org/rfc/rfc6749#section-3.1.2
+		if (getAppLocation() != null) {
+			sessionStorage.setItem("docLocation", document.location.hash);
+			setWindowURLFragment(null);
+		}
+		const keycloak = new Keycloak({
+			url: "https://app.please-open.it",
+			realm: "5d970145-a48d-4e9b-84ab-15de19f9d3a5",
+			clientId: window.location.hostname
+		});
+		await keycloak.init({
+			onLoad: "login-required",
+			flow: "implicit",
+			checkLoginIframe: false
+		});
+		let docLocation = sessionStorage.getItem("docLocation");
+		if (docLocation != null) {
+			sessionStorage.removeItem("docLocation");
+			setWindowURLFragment(docLocation);
+		}
+		return keycloak.token;
+	},
```

Comment:
The new login method uses the KeyCloak Javascript adapter to login to the IdP. The only complication here is that this app uses the URL fragment for identification of the TodoList (its subdomain name), which clashes with the fragments used when the IdP redirects back to the application. To avoid this problem we use session storage to re-instigate the original fragment after login.